mofa/simulation/lammps.py

```diff
@@ -8,9 +14,103 @@ class LAMMPSRunner:
+                              ff_string='UFF4MOF',
+                              small_molecule_force_field=None,
+                              outdir=lmp_path,
+                              charges=False,
+                              parallel=False,
+                              replication='2x2x2',
+                              read_cifs_pymatgen=True,
+                              add_molecule=None,
+                              small_molecule_file=None)
+            in_file_name = [x for x in os.listdir(lmp_path) if x.startswith("in.") and not x.startswith("in.lmp")][0]
+            data_file_name = [x for x in os.listdir(lmp_path) if x.startswith("data.") and not x.startswith("data.lmp")][0]
+            in_file_rename = "in.lmp"
+            data_file_rename = "data.lmp"
+            print("Reading data file for element list: " + os.path.join(lmp_path, data_file_name))
+            with io.open(os.path.join(lmp_path, data_file_name), "r") as rf:
+                df = pd.read_csv(io.StringIO(rf.read().split("Masses")[1].split("Pair Coeffs")[0]), sep=r"\s+", header=None)
+                element_list = df[3].to_list()
+            with io.open(os.path.join(lmp_path, in_file_rename), "w") as wf:
+                print("Writing input file: " + os.path.join(lmp_path, in_file_rename))
+                with io.open(os.path.join(lmp_path, in_file_name), "r") as rf:
+                    print("Reading original input file: " + os.path.join(lmp_path, in_file_name))
+                    wf.write(rf.read().replace(data_file_name, data_file_rename) + """
+
+# simulation
+
+fix                 fxnpt all npt temp 300.0 300.0 $(200.0*dt) tri 1.0 1.0 $(800.0*dt)
+variable            Nevery equal """ + "%d" % report_frequency + """
+
+thermo              ${Nevery}
+thermo_style        custom step cpu dt time temp press pe ke etotal density xlo ylo zlo cella cellb cellc cellalpha cellbeta cellgamma
+thermo_modify       flush yes
+
+minimize            1.0e-10 1.0e-10 10000 100000
+reset_timestep      0
+
+dump                trajectAll all custom ${Nevery} dump.lammpstrj.all id type element x y z q
+dump_modify         trajectAll element """ + " ".join(element_list) + """
+
+timestep            0.5
+run                 """ + "%d" % timesteps + """
+undump              trajectAll
+write_restart       relaxing.*.restart
+write_data          relaxing.*.data
+
+""")
+            os.remove(os.path.join(lmp_path, in_file_name))
+            shutil.move(os.path.join(lmp_path, data_file_name), os.path.join(lmp_path, data_file_rename))
+            print("Success!!\n\n")
+            return_code = 0
+
+        except Exception as e:
+            print(e)
+            print("Failed!! Removing files...\n\n")
+            shutil.rmtree(lmp_path)
+            return_code = -1
+
+        return lmp_path, return_code
 
     def run_molecular_dynamics(self, mof: ase.Atoms, timesteps: int, report_frequency: int) -> list[ase.Atoms]:
```

Comment:
From what I'm learning, we must pass the full CIF file because it contains the bonds (unlike the ase.Atoms object). I'm going to change this interface

Reply:
Nice! That's good to know, as it means we could avoid using the CIF format (which I find to be problematic)